hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,633 @@
+  {name: "sd[2]", wave: "xz.......3344552233z.22x", data: ["A[21]", "A[17]", "A[13]", "A[9]", "A[5]", "A[1]", "B[29]", "B[25]", "B[21]", "B[17]", "X[29]", "X[25]"]},
+  {name: "sd[3]", wave: "xz.......3344552233z.22x", data: ["A[20]", "A[16]", "A[12]", "A[8]", "A[4]", "A[0]", "B[28]", "B[24]", "B[20]", "B[16]", "X[28]", "X[24]"]},
+],
+   {name:""},
+   ["ByteOrder=1",
+  {name: "sd[0]", wave: "x555555554433225544z.55x", data: ["A[7]", "A[6]", "A[5]", "A[4]", "A[3]", "A[2]", "A[1]", "A[0]",
+                                                           "A[15]", "A[11]", "A[23]", "A[19]", "A[31]", "A[27]", "B[7]", "B[3]", "B[15]", "B[11]", "X[7]", "X[3]"]},
+  {name: "sd[1]", wave: "xz.......4433225544z.55x", data: ["A[14]", "A[10]", "A[22]", "A[18]", "A[30]", "A[26]", "B[6]", "B[2]", "B[14]", "B[10]", "X[6]","X[2]"]},
+  {name: "sd[2]", wave: "xz.......4433225544z.55x", data: ["A[13]", "A[9]", "A[21]", "A[17]", "A[29]", "A[25]", "B[5]", "B[1]", "B[13]", "B[9]", "X[5]", "X[1]"]},
+  {name: "sd[3]", wave: "xz.......4433225544z.55x", data: ["A[12]", "A[8]", "A[20]", "A[16]", "A[28]", "A[24]", "B[4]", "B[0]", "B[12]", "B[8]", "X[4]", "X[0]"]},
+  ],
+  ],
+  head: {
+    text: "Serial bit ordering for 6 bytes transmitted from FIFO words 'A[31:0]' and 'B[31:0]', and 1 byte received into word 'X[31:0]'",
+  },
+  foot: {
+    text: "Note: SPI_HOST command sent with COMMAND.TX1_CNT = 1, .TX4_CNT = 5, .DUMMY_CNT = 2, and  .RX_CNT=1"
+  }
+}
+{{< /wavejson >}}
+
+Beyond ignoring any unused bytes at the end of each transaction, there is no mechanism specified in this spec for supporting unaligned writes to TXDATA, or more generally for supporting byte-enables when writing to TXDATA.
+<!-- QFR: Does regtool actually support Byte enables? I took a look at the implementation and it seemed to me like it does not.-->
+The SPI transmission always starts with the MSB of TXDATA (or if `ByteOrder` = 1, the LSB).
+The process of copying command sequences from processor memory to TXDATA will be easiest when the address of the command is 4-byte aligned, and all transmitted data is adjacent in memory.
+
+Support for explicit byte-enables may come in a future revision.
+
+## Pass-through Mode
+
+The SPI_HOST also supports a special "Pass-through" mode, which allows for the direct control of the serial interface, by another block (namely SPI_DEVICE).
+
+To enable this mode, one sets {{< regref "CONTROL.PASSTHRU" >}} to one.
+
+A multiplexor then configures the output signals `cio_sck_o` and `cio_cs_no` to directly match the inter-module signals `pt_sck_i` and `pt_cs_ni[MaxCS-1:0]`.
+Likewise the bidirectional port-control `cio_sd_o[3:0]` and `cio_sdt_t[3:0]` are then respectively driven by inter-module signals `cio_sdo_i[3:0]` and `cio_sdi_o[3:0]`.
+Meanwhile the peripheral data input signal `cio_sdo_o[3:0]` is always connected to the inter-module `pt_sdi_o[3:0]`.
```

Comment:
@eunchan 
there may be cases where we need the control to revert from passthrough back to the normal spi host.  What is the right way to handle this? 

I'm not quire sure if we defined a hardware mechanism to change the control, or if it's okay to always leave it in software hands. It feels like leaving in software hands is OKAY to me, but we need to be a bit careful about what might happen if a passthrough transaction is ongoing and we suddenly change it.

Reply:
I'm more worried about the converse scenario, if there is a software driven command in progress and passthrough steals the bus. 

If this collision behavior is not well-defined, then I would greatly appreciate it if we could break this out into an issue, (to avoid holding this PR hostage.)  This could go in circles a while.

Reply:
yeah sounds good to break it out into an issue.

Reply:
Created Issue: https://github.com/lowRISC/opentitan/issues/5134